AAS V3.0 Classes and Attribute Identifiers (#31)
admin-shell-io/id: aas/API/3/0/README.md

# AAS (Asset Administration Shell)

> This is the sub-namespace reserved for the "Asset Administration Shell - Part 2 Application Programming Interfaces Version 3.0".

Reviewer: ```suggestion
This is the sub-namespace reserved for the "Specification of the Asset Administration Shell - Part 2: Application Programming Interfaces", Version 3.0 (IDTA-01002-3-0).
```

> The document 'Details of the Asset Administration Shell Part 2 - Application Programming Interfaces' [1] defines the meaning of its content and the used identifiers.

Reviewer: ```suggestion
The document 'Specification of the Asset Administration Shell - Part 2: Application Programming Interfaces', Version 3.0 (IDTA-01002-3-0) [1] defines the meaning of its content and the used identifiers.
```

## Introduction

> The content of this area is defined in the 'Details of the Asset Administration Shell -
> Part 2'. Only identifiers that are defined in this document are allowed in this

Reviewer: ```suggestion
The content of this area is defined in the "Specification of the Asset Administration Shell - Part 2: Application Programming Interfaces", Version 3.0 (IDTA-01002-3-0). Only identifiers that are defined in this document are allowed in this
```

## PostAllAssetLinksById (Operation)

[https://admin-shell.io/aas/API/PostAllAssetLinksById/3/0](https://admin-shell.io/aas/API/PostAllAssetLinksById/3/0): Creates or updates all Asset identifier key-value-pairs linked to an Asset Administration Shell to edit discoverable content

## DeleteAllAssetLinksById (Operation)

[https://admin-shell.io/aas/API/DeleteAllAssetLinksById/3/0](https://admin-shell.io/aas/API/DeleteAllAssetLinksById/3/0): Deletes all Asset identifier key-value-pair linked to an Asset Administration Shell

## Descriptor (Class)

[https://admin-shell.io/aas/API/DataTypes/Descriptor/3/0](https://admin-shell.io/aas/API/DataTypes/Descriptor/3/0): The self-describing information of a network resource.

## AssetAdministrationShellDescriptor (Class)

[https://admin-shell.io/aas/API/DataTypes/AssetAdministrationShellDescriptor/3/0](https://admin-shell.io/aas/API/DataTypes/AssetAdministrationShellDescriptor/3/0): Descriptor of an Asset Administration Shell

## SubmodelDescriptor (Class)

[https://admin-shell.io/aas/API/DataTypes/SubmodelDescriptor/3/0](https://admin-shell.io/aas/API/DataTypes/SubmodelDescriptor/3/0): A descriptor of a submodel

## Endpoint (Class)

> [https://admin-shell.io/aas/API/DataTypes/Endpoint/3/0](https://admin-shell.io/aas/API/DataTypes/Endpoint/3/0): The endpoint description of a network resource.

Reviewer: ```suggestion
[https://admin-shell.io/aas/API/DataTypes/Endpoint/3/0](https://admin-shell.io/aas/API/DataTypes/Endpoint/3/0): The endpoint description of a network resource
```

## ProtocolInformation (Class)

> [https://admin-shell.io/aas/API/DataTypes/ProtocolInformation/3/0](https://admin-shell.io/aas/API/DataTypes/ProtocolInformation/3/0): The protocol information of a network resource endpoint will be defined in DIN SPEC 16593-2.

Reviewer: ```suggestion
[https://admin-shell.io/aas/API/DataTypes/ProtocolInformation/3/0](https://admin-shell.io/aas/API/DataTypes/ProtocolInformation/3/0): The protocol information of a network resource endpoint
```

## MessageTypeEnum (Enumeration)

[https://admin-shell.io/aas/API/DataTypes/MessageTypeEnum/3/0](https://admin-shell.io/aas/API/DataTypes/MessageTypeEnum/3/0): The message type

## OperationRequest (Class)

[https://admin-shell.io/aas/API/DataTypes/OperationRequest/3/0](https://admin-shell.io/aas/API/DataTypes/OperationRequest/3/0): The operation request object

## OperationResult (Class)

[https://admin-shell.io/aas/API/DataTypes/OperationResult/3/0](https://admin-shell.io/aas/API/DataTypes/OperationResult/3/0): The operation’s invocation result object

## ExecutionState (Class)

[https://admin-shell.io/aas/API/DataTypes/ExecutionState/3/0](https://admin-shell.io/aas/API/DataTypes/ExecutionState/3/0): The operation’s invocation result state

## OperationHandle (Class)

[https://admin-shell.io/aas/API/DataTypes/OperationHandle/3/0](https://admin-shell.io/aas/API/DataTypes/OperationHandle/3/0): The returned handle of an operation’s asynchronous invocation used to request the current state of the operation’s execution.

## Contact

> The entries of this sub-namespace have been created by [Sebastian Bader](https://github.com/sebbader)

Reviewer: this is visible from history of committs. Shouldn't it be IDTA in this case?

Author: I think there wasn't an IDTA when I initially wrote this :-)

> on behalf of the IDTA Working Group 'Asset Administration Shell'.

Reviewer: ```suggestion
on behalf of the IDTA Working Group 'Specifications of the Asset Administration Shell'.
```

Contact via a [web form](https://industrialdigitaltwin.org/en/contact).

> [1] S. Bader, B. Berres, B. Boss, A. Graf Gatterburg, M. Hoffmeister et al.

Reviewer: There are not authors mentioned in the published document

> ['Details of the Asset Administration Shell Part 2: Application Programming Interfaces'](https://industrialdigitaltwin.org/en/content-hub/aasspecifications), Version 3.0,

Reviewer: ```suggestion
['Specification of the Asset Administration Shell - Part 2: Application Programming Interfaces'](https://industrialdigitaltwin.org/en/content-hub/aasspecifications), Version 3.0,
```
